main.js

```diff
@@ -158,7 +158,16 @@ $(function() {
         let storage = getStorage();
 
         storage.forEach(function(item) {
-          document.querySelector('#LC' + item.key.split('-')[0] + ' .collapser').click();
+          let [lower, upper] = item.key.split('-');
+          upper++;
+          let lowerLine = document.querySelector('#LC' + lower).textContent;
+          let upperLine = document.querySelector('#LC' + upper).textContent;
+
+          if (lowerLine.includes('{') && upperLine.includes('}')) {
```

Comment:
Using the the `pairs` map, which is the map of all the code blocks on the page with key -> values of lower -> upper line numbers. So something like this would work:
````javascript
if (pairs.get(lower) === higher) {
  // collapse the block 
}
````